pkg/controller/deployment/deployment_controller.go

```diff
@@ -566,19 +566,21 @@ func (dc *DeploymentController) getPodMapForDeployment(d *apps.Deployment, rsLis
 // syncDeployment will sync the deployment with the given key.
 // This function is not meant to be invoked concurrently with the same key.
 func (dc *DeploymentController) syncDeployment(key string) error {
-	startTime := time.Now()
-	klog.V(4).Infof("Started syncing deployment %q (%v)", key, startTime)
-	defer func() {
-		klog.V(4).Infof("Finished syncing deployment %q (%v)", key, time.Since(startTime))
-	}()
-
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
+		klog.ErrorS(err, "Failed to split meta namespace cache key", "cacheKey", key)
 		return err
 	}
+
+	startTime := time.Now()
+	klog.V(4).InfoS("Started syncing deployment", "deployment", klog.KRef(namespace, name), "startTime", startTime)
+	defer func() {
+		klog.V(4).InfoS("Finished syncing deployment", "deployment", klog.KRef(namespace, name), "duration", time.Since(startTime))
+	}()
```

Comment:
Why moving this?

Reply:
per https://github.com/kubernetes/kubernetes/pull/97507#discussion_r575173518 `We should avoid passing different values into same logging key, deployment is reserved for using klog.KObj or klog.KRef`.

I moved those line to be able to use `klog.KRef(namespace, name)` to write deployment name instead of cacheKey.

Reply:
kk